service/main.go

```diff
@@ -0,0 +1,172 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/LeonidVas/tarantool-supervisor/service/api/supervisorhttp"
+	"github.com/LeonidVas/tarantool-supervisor/service/supervisor"
+)
+
+// args describes the parsed arguments.
+type args struct {
+	//CfgPath - path to Supervisor config.
+	CfgPath string
+	// Host - address to start the HTTP server.
+	Host string
+}
+
+// parseArgs returns the parsed arguments.
+func parseArgs() *args {
+	var args args
+	flag.StringVar(&args.CfgPath, "cfg", "cfg.json",
+		"path to Supervisor config.")
+	flag.StringVar(&args.Host, "host", "127.0.0.1:8080",
+		"address to start the HTTP server(addr:port).")
+	flag.Parse()
+
+	return &args
+}
+
+// parseCfg parse the Supervisor JSON config.
+func parseCfg(path string) (*supervisor.Cfg, error) {
+	// Check is the file exists.
+	if _, err := os.Stat(path); err != nil {
+		return nil, err
+	}
+
+	// Open the file.
+	jsonFile, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer jsonFile.Close()
+
+	// Set defaults.
+	cfg := supervisor.Cfg{InstancesDir: "/etc/tarantool/supervisor/instances",
+		TermTimeout: 30}
+
+	// Read and parse config.
+	byteValue, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		return nil, err
+	}
+	if err = json.Unmarshal(byteValue, &cfg); err != nil {
+		return nil, err
+	}
+
+	// In the config, the time is indicated in seconds. Convert the value.
+	cfg.TermTimeout = cfg.TermTimeout * time.Second
+
+	return &cfg, nil
+}
+
+// handleSigChld handles the SIGCHLD signal.
+func handleSigChld(sv *supervisor.Supervisor) {
+	// Get PID of the terminated Instance.
+	pid, err := syscall.Wait4(-1, nil, syscall.WNOHANG, nil)
+	if err != nil || pid == 0 {
+		// If the error "no child processes" (syscall.Errno(10)) or pid == 0 -
+		// this means that the process was stopped as planned, and the wait ()
+		// function has already been called.
+		if !errors.Is(err, syscall.Errno(10)) && pid != 0 {
+			log.Printf(`Can't get the PID of a process by using "Wait4". Error: "%v"`,
+				err)
+		}
+		return
+	}
+	log.Printf("pid: %v\terr: %v", pid, err)
+
+	// If the isRestart Instance flag is true, try to restart the Instance.
+	if id, err := sv.RestartAfterTermInstance(pid); err != nil {
```

Comment:
https://github.com/LeonidVas/tarantool-supervisor/blob/lvasiliev/MVP/supervisor/core/supervisor.go#L127

Reply:
Then in case of `Restarted=false` the error would be `Can't restart the Instance: The IsRestart Instance flag is false`. But if user set `Restarted=false` he doesn't expect instance to be restarted.

Reply:
Yep. But I look at it as a failed restart attempt (for a function). The reason will be correctly displayed in the error and the event will be saved to the log.